spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/security/LdapAuthenticationConfiguration.java

```diff
@@ -0,0 +1,177 @@
+import org.springframework.security.config.annotation.authentication.configurers.ldap.LdapAuthenticationProviderConfigurer;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.ldap.userdetails.LdapAuthoritiesPopulator;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+* A security configuration that conditionally sets up in-memory users from a file.
+*
+* @author Marius Bogoevici
+* @author Gunnar Hillert
+*
+* @since 1.1.0
+*/
+@Configuration
+@ConfigurationProperties(prefix = "dataflow.security.authentication.ldap")
+@ConditionalOnProperty("dataflow.security.authentication.ldap.enabled")
+public class LdapAuthenticationConfiguration extends GlobalAuthenticationConfigurerAdapter {
+
+	private String url;
+
+	private String userDnPattern;
+
+	private String managerDn;
+
+	private String managerPassword;
+
+	private String userSearchBase = "";
+
+	private String userSearchFilter;
+
+	private String groupSearchFilter = "";
+
+	private String groupSearchBase = "";
+
+	private String groupRoleAttribute = "cn";
+
+	public String getUrl() {
+		return url;
+	}
+
+	public void setUrl(String url) {
+		this.url = url;
+	}
+
+	public String getUserDnPattern() {
+		return userDnPattern;
+	}
+
+	public void setUserDnPattern(String userDnPattern) {
+		this.userDnPattern = userDnPattern;
+	}
+
+	public String getManagerDn() {
+		return managerDn;
+	}
+
+	public void setManagerDn(String managerDn) {
+		this.managerDn = managerDn;
+	}
+
+	public String getManagerPassword() {
+		return managerPassword;
+	}
+
+	public void setManagerPassword(String managerPassword) {
+		this.managerPassword = managerPassword;
+	}
+
+	public String getUserSearchBase() {
+		return userSearchBase;
+	}
+
+	public void setUserSearchBase(String userSearchBase) {
+		this.userSearchBase = userSearchBase;
+	}
+
+	public String getUserSearchFilter() {
+		return userSearchFilter;
+	}
+
+	public void setUserSearchFilter(String userSearchFilter) {
+		this.userSearchFilter = userSearchFilter;
+	}
+
+	public String getGroupSearchFilter() {
+		return groupSearchFilter;
+	}
+
+	public void setGroupSearchFilter(String groupSearchFilter) {
+		this.groupSearchFilter = groupSearchFilter;
+	}
+
+	public String getGroupSearchBase() {
+		return groupSearchBase;
+	}
+
+	public void setGroupSearchBase(String groupSearchBase) {
+		this.groupSearchBase = groupSearchBase;
+	}
+
+	public String getGroupRoleAttribute() {
+		return groupRoleAttribute;
+	}
+
+	public void setGroupRoleAttribute(String groupRoleAttribute) {
+		this.groupRoleAttribute = groupRoleAttribute;
+	}
+
+	@Override
+	public void init(AuthenticationManagerBuilder auth) throws Exception {
+
+		LdapAuthenticationProviderConfigurer<AuthenticationManagerBuilder> ldapConfigurer = auth.ldapAuthentication();
+
+		Assert.hasText(url, "'url' must not be empty");
```

Comment:
Can we move the properties in their own class and/or add jsr303 validation to them?